Readme.md

```diff
@@ -1,125 +1,86 @@
-
-
-```js
-       function Grid_BatchEditStartEditing(s, e) {
-            var templateColumn = s.GetColumnByField("C1");
-            if (!e.rowValues.hasOwnProperty(templateColumn.index))
-                return;
-            var cellInfo = e.rowValues[templateColumn.index];
-            C1spinEdit.SetValue(cellInfo.value);
-            if (e.focusedColumn === templateColumn)
-                C1spinEdit.SetFocus();
-        }
-
-```
-
-
-<p> </p>
-<p>3. Handle the <a href="https://help.devexpress.com/#AspNet/DevExpressWebASPxGridViewScriptsASPxClientGridView_BatchEditEndEditingtopic">BatchEditEndEditing</a> event to pass a value entered in the editor to the grid's cell:</p>
-
-
-```js
-        function Grid_BatchEditEndEditing(s, e) {
-            var templateColumn = s.GetColumnByField("C1");
-            if (!e.rowValues.hasOwnProperty(templateColumn.index))
-                return;
-            var cellInfo = e.rowValues[templateColumn.index];
-            cellInfo.value = C1spinEdit.GetValue();
-            cellInfo.text = C1spinEdit.GetText();
-            C1spinEdit.SetValue(null);
-        }
-
-```
-
-
-<p> </p>
-<p>4. The <a href="https://help.devexpress.com/#AspNet/DevExpressWebASPxGridViewScriptsASPxClientGridView_BatchEditRowValidatingtopic">BatchEditRowValidating</a> event allows validating the grid's cell based on the entered value:</p>
-
-
-```js
-       function Grid_BatchEditRowValidating(s, e) {
-            var templateColumn = s.GetColumnByField("C1");
-            var cellValidationInfo = e.validationInfo[templateColumn.index];
-            if (!cellValidationInfo) return;
-            var value = cellValidationInfo.value;
-            if (!ASPxClientUtils.IsExists(value) || ASPxClientUtils.Trim(value) === "") {
-                cellValidationInfo.isValid = false;
-                cellValidationInfo.errorText = "C1 is required";
-            }
-        }
-
-```
-
-
-<p> </p>
-<p>5. Finally, handle the editor's client-side <a href="https://documentation.devexpress.com/AspNet/DevExpressWebASPxEditorsScriptsASPxClientTextEdit_KeyDowntopic.aspx">KeyDown</a> and <a href="https://documentation.devexpress.com/AspNet/DevExpressWebASPxEditorsScriptsASPxClientEdit_LostFocustopic.aspx">LostFocus</a> events to emulate the behavior of standard grid editors when an end-user uses a keyboard or mouse:</p>
+5. Handle the editor's client-side `KeyDown` and `LostFocus` events to emulate the editor behavior when a user presses a key or clicks outside the editor.
 
+> **Note**  
+> When you implement an edit item template, the control does not update the data source automatically. Handle the grid's server-side [RowUpdating](https://docs.devexpress.devx/AspNet/DevExpress.Web.ASPxGridView.RowUpdating), [RowInserting](https://docs.devexpress.devx/AspNet/DevExpress.Web.ASPxGridView.RowInserting), and [RowDeleting](https://docs.devexpress.devx/AspNet/DevExpress.Web.ASPxGridView.RowDeleting) events to manually update the data source.
```

Comment:
> to **manually** update the data source.

Word order. "manually" is a complement that describes circumstances and answers the question "how?" The complement should follow the main sentence part, not break it in two. You did it right first when said

> the control does not update the data source **automatically**.

```suggestion
> When you implement an edit item template, the control does not update the data source automatically. Handle the grid's server-side [RowUpdating](https://docs.devexpress.devx/AspNet/DevExpress.Web.ASPxGridView.RowUpdating), [RowInserting](https://docs.devexpress.devx/AspNet/DevExpress.Web.ASPxGridView.RowInserting), and [RowDeleting](https://docs.devexpress.devx/AspNet/DevExpress.Web.ASPxGridView.RowDeleting) events to update the data source manually.
```